app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package co.nimblehq.compose.crypto.ui.screens.home
+
+import co.nimblehq.compose.crypto.domain.usecase.GetMyCoinsUseCase
+import co.nimblehq.compose.crypto.ui.base.*
+import co.nimblehq.compose.crypto.ui.uimodel.CoinItemUiModel
+import co.nimblehq.compose.crypto.ui.uimodel.toUiModel
+import co.nimblehq.compose.crypto.util.DispatchersProvider
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.*
+import javax.inject.Inject
+
+private const val MY_COINS_CURRENCY = "usd"
+private const val MY_COINS_ORDER = "market_cap_desc"
+private const val MY_COINS_PRICE_CHANGE_IN_HOUR = "24h"
+private const val MY_COINS_ITEM_PER_PAGE = 10
+
+interface Output : BaseOutput {
+    val myCoins: StateFlow<List<CoinItemUiModel>>
+}
+
+@HiltViewModel
+class HomeViewModel @Inject constructor(
+    dispatchers: DispatchersProvider,
+    private val getMyCoinsUseCase: GetMyCoinsUseCase
```

Comment:
I don't think we need `private val`, if we will call it only in `init` scope 🤔 
```suggestion
    getMyCoinsUseCase: GetMyCoinsUseCase
```